Drv/Ip/SocketComponentHelper.cpp

```diff
@@ -29,60 +27,54 @@ void SocketComponentHelper::start(const Fw::StringBase &name,
                                      const Os::Task::ParamType stack,
                                      const Os::Task::ParamType cpuAffinity) {
     FW_ASSERT(m_task.getState() == Os::Task::State::NOT_STARTED);  // It is a coding error to start this task multiple times
-    FW_ASSERT(not this->m_stop);        // It is a coding error to stop the thread before it is started
+    this->m_stop = false;
     m_reconnect = reconnect;
     // Note: the first step is for the IP socket to open the port
     Os::Task::Arguments arguments(name, SocketComponentHelper::readTask, this, priority, stack, cpuAffinity);
     Os::Task::Status stat = m_task.start(arguments);
-    FW_ASSERT(Os::Task::OP_OK == stat, static_cast<NATIVE_INT_TYPE>(stat));
-}
-
-SocketIpStatus SocketComponentHelper::startup() {
-    this->m_started = true;
-    return this->getSocketHandler().startup();
-}
-
-bool SocketComponentHelper::isStarted() {
-    bool is_started = false;
-    is_started = this->m_started;
-    return is_started;
+    FW_ASSERT(Os::Task::OP_OK == stat, static_cast<FwAssertArgType>(stat));
 }
 
 SocketIpStatus SocketComponentHelper::open() {
     SocketIpStatus status = SOCK_FAILED_TO_GET_SOCKET;
-    NATIVE_INT_TYPE fd = -1;
-    this->m_lock.lock();
-    if (not this->m_open) {
-        FW_ASSERT(this->m_fd == -1 and not this->m_open); // Ensure we are not opening an opened socket
-        status = this->getSocketHandler().open(fd);
-        this->m_fd = fd;
+    OpenState local_open = OpenState::OPEN;
+    // Scope to guard lock
+    {
+        Os::ScopeLock scopeLock(m_lock);
+        if (this->m_open == OpenState::NOT_OPEN) {
+            this->m_open = OpenState::OPENING;
+        }
```

Comment:
else {
    local_open = SKIP
}

Reply:
SUCCESS == successfully opened socket, or another thread is doing so on your behalf.

Reply:
instead of success, return an error ("already opening") and allow that in `reconnect`